### PR TITLE
Fixes dedupe exception when dupe metadata

### PR DESCRIPTION
**Pull request by bkase**
Log the dupes in a special key

### CODE REVIEW

File: src/lib/logger/impl.ml [hunk 1]
```diff
@@ -324,13 +324,23 @@ let change_id {null; metadata; id= _} ~id = {null; metadata; id}
...
 let make_message (t : t) ~level ~module_ ~location ~metadata ~message ~event_id
     =
+  let global_metadata' =
+    let m = !global_metadata in
+    let key_cmp (k1, _) (k2, _) = String.compare k1 k2 in
+    match List.find_all_dups m ~compare:key_cmp with
+    | [] ->
+        m
+    | dups ->
+        ("$dup", `List (List.map ~f:(fun (s, _) -> `String s) dups))
```
Comment (mrmr1993) [inline src/lib/logger/impl.ml:334]: It would be nice to log the metadata for all of the dups here, but that can be a follow-up PR

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/logger/impl.ml (hunk 1 above)
Comment (psteckler) [inline src/lib/logger/impl.ml:334]: can we make this "duplicated_items", to avoid jargon?

File: src/lib/logger/impl.ml [hunk 2]
```diff
@@ -324,13 +324,23 @@ let change_id {null; metadata; id= _} ~id = {null; metadata; id}
...
 let make_message (t : t) ~level ~module_ ~location ~metadata ~message ~event_id
     =
+  let global_metadata' =
+    let m = !global_metadata in
+    let key_cmp (k1, _) (k2, _) = String.compare k1 k2 in
+    match List.find_all_dups m ~compare:key_cmp with
+    | [] ->
+        m
+    | dups ->
+        ("$duplicated_keys", `List (List.map ~f:(fun (s, _) -> `String s) dups))
```
Comment (psteckler) [inline src/lib/logger/impl.ml:334]: should that $ be there?

Comment (bkase) [reply src/lib/logger/impl.ml:334]: Yes, this is there to denote "machine-generated" (this is a convention in JavaScript / JSON)

**Pull request merged**